src/agents.ts

```diff
@@ -16,19 +16,28 @@
 
 import {Agent as HTTPAgent} from 'http';
 import {Agent as HTTPSAgent} from 'https';
+import {parse} from 'url';
 import {Options} from './';
 
-const pool = new Map<string, HTTPAgent>();
+const createHttpProxyAgent = require('http-proxy-agent');
```

Comment:
We _really_ want to keep the deferred loading here.  Those modules create slowdowns at process start time, and are rarely used.  Only in cases where customers need to configure a proxy server.